keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/handlers/SlackNotificationHandler.kt

```diff
@@ -0,0 +1,28 @@
+package com.netflix.spinnaker.keel.slack.handlers
+
+import com.netflix.spinnaker.keel.notifications.NotificationType
+import com.netflix.spinnaker.keel.slack.SlackNotificationEvent
+
+/**
+ * Implement this interface to send different types of slack notifications. Each notification is being consturct by [constructMessage]
+ * See: [PinnedNotificationHandler] for example
+ */
+
+interface SlackNotificationHandler <T: SlackNotificationEvent> {
+  val type: NotificationType
+
+  fun constructMessage(notification: T)
```

Comment:
This is not only constructing the message, but sending it out. Maybe `constructAndSendMessage`, or just `sendMessage`?

Reply:
sounds good!